bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -298,8 +305,25 @@ private void write(Page page, XMLStreamWriter stream, ResourceResolver resolver)
         stream.writeEndElement();
     }
 
-    private boolean isHidden(final Page page) {
-        return page.getProperties().get(this.excludeFromSiteMapProperty, false);
+    private boolean isHiddenOrProtected(Page page) {
+        boolean flag= false;
+        if(this.excludeFromSiteMapProperty != null){
+            String [] multipleProperties = this.excludeFromSiteMapProperty.split(",");
+            for (String prop: multipleProperties) {
+                flag = flag || page.getProperties().get(prop, Boolean.FALSE);
+            }
+        }
+        return flag;
+    }
+
+    private boolean isExcluded(Page page){
+        boolean flag = false;
+        if(this.template != null){
+            for(String pageTemplate : this.template){
+                flag = flag || page.getProperties().get("cq:template", String.class).equalsIgnoreCase(pageTemplate);
```

Comment:
```suggestion
                flag = flag || page.getProperties().get("cq:template", "").equalsIgnoreCase(pageTemplate);
```

Changing default value in `.get(.., "")` to empty string to avoid potential NPE.